src/XMakeBuildEngine/Xml/ProjectXmlUtilities.cs

```diff
@@ -48,6 +48,13 @@ private static List<XmlElementWithLocation> GetChildElements(XmlElementWithLocat
                         VerifyThrowProjectValidNamespace(childElement);
                         children.Add(childElement);
                         break;
+                    case XmlNodeType.Text:
+                        // Whitespace is read as a #text element so if the node is purely whitespace, just ignore it.
+                        if (!String.IsNullOrWhiteSpace(child.InnerText) && throwForInvalidNodeTypes)
+                        {
+                            ThrowProjectInvalidChildElement(child.Name, element.Name, element.Location);
+                        }
```

Comment:
I'd prefer

```C#
if (string.IsNullOrWhitespace(child.InnerText)) {
    // long whitespace comes back as text, see [link]
    break;
}
goto default;
```

and rely on the `default` case, instead of having to check whether or not to throw and throwing in this case.
with a fall